chall06/dperez-z.c

```diff
@@ -0,0 +1,246 @@
+#include <stdio.h>
+#include <unistd.h>
+#include <stdlib.h>
+
+int check_is_king(char *board)
+{
+	int cont;
+	int nking;
+
+	nking=0;
+	cont = 0;
+	while(board[cont]!='\0')
+	{
+		if(board[cont]=='K')
+			nking++;
+		cont++;
+	}
+	if (nking==1)
+		return(1);
+	return(0);
+}
+
+int check_is_pieces(char *board)
+{
+        int cont;
+
+        cont = 0;
+        while(board[cont]!='\0')
+        {
+                if(board[cont]!='K' && board[cont]!='\0' && board[cont]!='.' && board[cont]!='P' && board[cont]!='R' && board[cont]!='Q' && board[cont]!='B' && board[cont]!='\n')
+                        return(0);
+                cont++;
+        }
+        return(1);
+}
+int check_dimension(char *board)
+{
+	int col;
+	int i;
+	int colmax;
+	int row;
+	int flag;
+
+	i = 0;
+	flag = 0;
+	row = 0;
+	col = 0;
+	colmax = 0;
+	while (board[i]!='\0')
+	{
+		if (board[i]=='\n')
+		{
+			if (flag == 0)
+			{
+				colmax = col;
+				flag = 1;
+			}
+			row ++;
+			if (i+1 != (row*colmax+row))
+				return(-1);
+		}
+		col++;	
+		i++;
+	}
+	if (row != colmax)
+		return(-1);
+	return(colmax);
+}
+
+int ft_strlen(char *board)
+{
+	int i;
+	i = 0;
+	while(board[i]!='\0')
+		i++;
+	return(i);
+}
+
+int check_rook(char *board, int row, int king)
+{
+	int i;
+	int n;
+
+	i = king;
+	n = 0;
+	while(i > 0 && i < ft_strlen(board))
+	{
+		i = i - n * row;
+		if (board[i]=='Q')
+			return(1);
+		n++;
+	}
+	i = king;
+	n = 0;
+	while(i > 0 && i < ft_strlen(board))
+        {
+                i = i +  n * row;
+                if (board[i]=='Q')
+                        return(1);
+                n++;
+        }
+
+
+	return(0);
+}
+
+				
+int ft_check_mate(char *board)
+{
+	int col;
```

Comment:
This is... not pretty. This exercise was asking for a coord based struct, I know there's no norminette, but this is much harder to read.